app/models/generic_object.rb

```diff
@@ -1,9 +1,9 @@
 class GenericObject < ApplicationRecord
   belongs_to :generic_object_definition
-  has_many   :custom_attributes, :as => :resource, :dependent => :destroy, :autosave => true
 
-  validates :name, :presence => true, :uniqueness => true
-  validate  :must_be_defined_attributes
+  has_many :custom_attributes, :as => :resource, :dependent => :destroy, :autosave => true
+  private  :custom_attributes
+  private  :custom_attributes=
```

Comment:
You can put these privates on the same line...there are a number of other custom_attributes derived methods as well, but it's up to you to make them private or not.